client/src/api.ts

```diff
@@ -97,6 +98,94 @@ interface GraphQLResponse<T> {
   data?: T;
 }
 
+export async function fetchMentorByAccessToken(
+  accessToken: string,
+  subject?: string,
+  topic?: string,
+  status?: string
+): Promise<Mentor> {
+  const headers = { Authorization: `bearer ${accessToken}` };
+  const result = await axios.post(
+    GRAPHQL_ENDPOINT,
+    {
+      query: `
+      query {
+        me {
+          mentor {
+            _id
+            name
+            firstName
+            title
+            mentorType
+            lastTrainedAt
+            defaultSubject {
+              _id
+            }
+            subjects {
+              _id
+              name
```

Comment:
does mentor-client use or display subject names at all? if not then this isn't needed

Reply:
Yeah, it does!